docs/docs/06-tools/03-debugging.md

```diff
@@ -0,0 +1,31 @@
+---
+title: Debugging
+id: Debugging
+description: Learn how to debug your Wing application
+keywords: [debugging, debug, test, vscode]
+---
+
+## Overview
+
+Internally Wing uses JavaScript to execute preflight and `inflight` code, so standard JavaScript debugging tools can be used to debug your Wing application. The best-supported debugger is the built-in VS Code one so this guide will focus on that. 
+
+### Local/Simulator Debugging
+
+To start, open your .w file in VS Code and set a breakpoint by clicking in the gutter to the left of the line number. Breakpoints can also be set in extern files. There are several ways to start the debugger, but let's use the "JavaScript Debug Terminal".
+Open the command palette and type "Debug: Open JavaScript Debug Terminal". This works for any wing commands like `wing test` and `wing it`, although keep in mind that `wing compile` will only debug preflight code.
+
+### Limitations
+
+- When using the Wing Console (`wing it`) and attempting to debug `inflight` code in a `test` or Function, the first execution of the test will not hit a breakpoint and will need to be run again
```

Comment:
I use inflight over "inflight" pretty often because it's a keyword. I would do the same with let or class. Should inflight specifically not use "`" or should other keywords also not get backticks?

Reply:
I think we should use backticks when we refer to the keyword and no backticks when we refer to the concept or the term.

For example: 

- "Let's declare a class that represents an employee."

- "To declare a data shape use the `struct` keyword followed by the name of the struct"

- "This value is a constant"

Makes sense?

Reply:
To me, the keyword is the concept and vice-versa. It helps my brain establish the link between the two, so that when I see the `word` in code I'll think of the concept and when I read the `word` in docs I'll think about how it's used in code.

I don't get confused when I see alternating `struct` vs "struct" so always using `backticks` isn't crazy important to me, but I'm not sold on the reasoning to avoid them so far.

The only reason I do get is aesthetics, because `keywords` can be pretty ugly when overused. So if that's part of our "Style Guide" then ok.